unpack-trees.c

```diff
@@ -665,17 +665,26 @@ static void mark_ce_used_same_name(struct cache_entry *ce,
 	}
 }
 
-static struct cache_entry *next_cache_entry(struct unpack_trees_options *o)
+static struct cache_entry *next_cache_entry(struct unpack_trees_options *o, int *hint)
 {
 	const struct index_state *index = o->src_index;
 	int pos = o->cache_bottom;
 
+	if (hint && *hint > pos)
+		pos = *hint;
```

Comment:
It looks like `hint` is never `NULL`, so the `if (hint)` checks can be removed.

You are right to keep the `if (*hint > pos)` here, since we want to use `cache_bottom` when possible.

Reply:
Good point - I was going to make it optional, but there wasn't really a downside to including it in other usage of `next_cache_entry`. Removed the null checks!